Current code 2021_09_08

VW71superbeetle opened the pull request:
Current Code

VW71superbeetle committed:
Login tweaks

Adjust login forms to have proper text types, and additional fields for register

jonguertin committed:
start admin area

VW71superbeetle committed:
Fix camelCase

VW71superbeetle committed:
Update Footer.jsx

Fix unsupported h7 tag

VW71superbeetle committed:
Order/OrderProducts support

Create REST API calls for orders and order_products. Expand returned data for getOrderProductByID

VW71superbeetle committed:
Cleanup Reviews functions

Cleanup reviews functions, and add missing call.

VW71superbeetle committed:
Login/Register changes

Fix login and register components, and bug fixes on user routes

VW71superbeetle committed:
Update .gitignore

Add .env to .gitignore

VW71superbeetle committed:
Code Cleanup

VW71superbeetle committed:
Health check

Checking server health

VW71superbeetle committed:
Update users.js

updating unsaved file from earlier.

jonguertin committed:
Update order_products.js

jonguertin committed:
Merge pull request #52 from 2103-USD/FrontEndAPIs_20210904

API Calls wrap-up 20210904

jonguertin committed:
added get all users and added id to users query

VW71superbeetle committed:
Merge branch 'dev' into GetAllUsers_20210904

VW71superbeetle committed:
Merge pull request #53 from 2103-USD/GetAllUsers_20210904

Get all users 20210904

jonguertin committed:
adding admin order card

jonguertin committed:
Merge branch 'dev' into AdminArea_20210904

VW71superbeetle committed:
Update User handling

Fix user handling so that login/register saves user object, and have NavBar adjust itself accordingly.
Also sets schoolboy/girl users to be non-admin.

jonguertin committed:
added admin orders, updated order card

VW71superbeetle committed:
Minor Code Cleanup

fix app Title, and remove unneeded auth functions.

VW71superbeetle committed:
Update Orders db calls

Update orders db calls to include user info.

VW71superbeetle committed:
Merge pull request #54 from 2103-USD/AdminArea_20210904

Admin area 20210904

jonguertin committed:
Merge pull request #55 from 2103-USD/AdminWork_20210906

Admin work 20210906

VW71superbeetle committed:
new getSpecificUser function

comment out line causing break, add support for getSpecificUser

demedukd committed:
Merge pull request #56 from 2103-USD/AdminWork_20210906

new getSpecificUser function

demedukd committed:
More products and frontend changes

demedukd committed:
Updated CSS

demedukd committed:
Footer, User.jsx, Button className

VW71superbeetle committed:
Merge branch 'dev' into DatabaseObjects20210904

VW71superbeetle committed:
Merge pull request #57 from 2103-USD/DatabaseObjects20210904

Database objects20210904

VW71superbeetle committed:
Various small fixes

fix breaking issue in orders, and fix login buttons

jonguertin committed:
Merge pull request #58 from 2103-USD/AdminWork_20210906

Various small fixes

demedukd committed:
slidetoscroll

demedukd committed:
Quantity buttons

VW71superbeetle committed:
Merge pull request #59 from 2103-USD/DatabaseObjects_09062021

Database objects 09062021

demedukd committed:
Fixed index.html, updated products layout

VW71superbeetle committed:
Merge pull request #60 from 2103-USD/DatabaseObjects_09062021

Fixed index.html, updated products layout

VW71superbeetle committed:
Adjust user handling

Move JWT secret to work around unknown issue.
Fix return for getAllUsers DB call.
console log cleanups.

demedukd committed:
Update inStock to int data type

. . . to track inventory.

VW71superbeetle committed:
Merge pull request #61 from 2103-USD/DatabaseObjects_20210908

Update inStock to int data type

jdmann commented on routes/orders.js:
```diff
@@ -20,67 +20,67 @@ const {
         }
         else {
-            res.status(500)
+            res.status(404)
```
might be better as 500

jdmann commented on src/components/AdminOrder.jsx:
```diff
@@ -0,0 +1,8 @@
+import React, { useEffect, useState } from "react";
+import { getAllOrders } from "../api";
+import "./style.css";
```
remove style.css ?

jdmann commented on src/components/AdminOrders.jsx:
```diff
@@ -0,0 +1,27 @@
+
+    return (
+        <div className="admin-orders">
+            <div>
```
good use case for a CSS Grid

jdmann commented on src/components/Register.jsx:
```diff
@@ -28,22 +32,56 @@ export const Register = ({ setUser }) => {
                     name="username"
                     value={form.username}
                     onChange={handleInput}
+                    autoComplete = "username"
```
native HTML validations are useful... required={true}

jdmann commented on src/components/Register.jsx:
```diff
@@ -1,25 +1,29 @@
+    const history = useHistory();
 
     const handleSubmit = async (e) => {
         e.preventDefault();
```
can also do a front-end validation here.... 
if (!form.username || form.password .... etc.) { return alert('missing fields');}

VW71superbeetle committed:
Admin User functionality

Adding Admin users page, and ability to edit a user account.

VW71superbeetle committed:
Admin Products feature

Create new Admin Products management pages.

VW71superbeetle committed:
Quick user fixes

VW71superbeetle committed:
Merge branch 'dev' into AdminWork_20210906

VW71superbeetle committed:
Update users.js

Fix broken login change from prior commit

VW71superbeetle committed:
inStock tweaks

Set inStock to have a default qty of 0, and editProducts page to display as numbers.

captnjak committed:
Merge pull request #63 from 2103-USD/AdminWork_20210906

Admin work 20210908

VW71superbeetle committed:
More sample data

Added order_products, refactored seed data into separate file.

VW71superbeetle committed:
Admin Orders

Admin orders have been built out.

demedukd committed:
Merge pull request #64 from 2103-USD/AdminWork_20210909

Admin work 20210909

captnjak committed:
updated CSS styling

captnjak committed:
updated Featured Items css

VW71superbeetle committed:
Merge pull request #65 from 2103-USD/FrontEnd9-9

Front end9 9

VW71superbeetle committed:
new Cart backend concept

Cart is now retrieved (or created) upon login/register. a logged in user always has a currentCart value saved in localStorage.
Fixed bugs in DB calls.

VW71superbeetle committed:
Redirect to home upon logou

Redirect user to home page upon logout.

captnjak committed:
Merge pull request #66 from 2103-USD/AdminWork_20210910

Backend cart work

VW71superbeetle committed:
Unique idx to orderProducts

Add unique index to order products, and on a conflict, the quantity is increased by 1

VW71superbeetle committed:
Code Cleanup

Remove unneeded declarations, and fix function name.

VW71superbeetle committed:
Empty Cart function

new function to empty a cart.

VW71superbeetle committed:
Add reviews and stars to products

Added reviews and star ratings to products.

captnjak committed:
Merge pull request #67 from 2103-USD/AdminWork_20210910

Admin work 20210910

VW71superbeetle committed:
Bug fixes

quick bug fixes

captnjak committed:
Merge pull request #68 from 2103-USD/AdminWork_20210910

Bug fixes

VW71superbeetle committed:
Add isActive for user account

Added new logic to handle disabling a user account.

captnjak committed:
add to cart features updated

VW71superbeetle committed:
Remove unused component

Remove unused AdminOrderCard component

VW71superbeetle committed:
Cleanup unused calls

Cleanup unused calls

demedukd committed:
Merge pull request #69 from 2103-USD/FrontEnd9-11

add to cart features updated

VW71superbeetle committed:
Update auth.js

set baseURL back to heroku

VW71superbeetle committed:
Merge branch 'dev' into AdminWork_20210910

demedukd committed:
Merge pull request #71 from 2103-USD/AdminWork_20210910

Admin work 20210910

captnjak committed:
issues with backend on removing product

VW71superbeetle committed:
Merge pull request #72 from 2103-USD/FrontEnd9-11

issues with backend on removing product

VW71superbeetle committed:
Remove price from updateProducts

We are not priceline -- you don't name your own price.

VW71superbeetle committed:
Code Cleanup

Removing console logs and warnings from console

VW71superbeetle committed:
Fix verifyUserIsOrderProductOwner

Fix verifyUserIsOrderProductOwner

VW71superbeetle committed:
Update auth.js

Fix optional helper functions

captnjak committed:
completed cart functionality

VW71superbeetle committed:
Merge pull request #74 from 2103-USD/FrontEnd9-11

completed cart functionality

captnjak committed:
Merge pull request #73 from 2103-USD/OrdersWork_20210911

Orders work 20210911

captnjak committed:
css overhaul

VW71superbeetle committed:
Stripe

Initial stripe integration. Key needs to be added into your .env file
Also added a Total into getCartByUser DB function

VW71superbeetle committed:
Merge pull request #75 from 2103-USD/FrontEnd9-11.2

css overhaul

captnjak committed:
Merge branch 'dev' into OrdersWork_20210911

demedukd committed:
Images, Filtering (half working)

Need to finish filters ... tomorrow ?

captnjak committed:
Merge pull request #76 from 2103-USD/OrdersWork_20210911

Stripe

captnjak committed:
Merge branch 'dev' into DatabaseObjects_20210911

captnjak committed:
Merge pull request #77 from 2103-USD/DatabaseObjects_20210911

Images, Filtering (half working)

demedukd committed:
Finished Filter!

captnjak committed:
Merge pull request #78 from 2103-USD/DatabaseObjects_20210912

Finished Filter!

demedukd committed:
Fixed Filter, removed dead and unborn code.

captnjak committed:
Merge pull request #79 from 2103-USD/DatabaseObjects_20210912

Fixed Filter, removed dead and unborn code.

demedukd committed:
NavBar

Ready for one more Heroku push? 😅

VW71superbeetle committed:
Merge pull request #80 from 2103-USD/DatabaseObjects_20210912

VW71superbeetle committed:
Add support for Toastify

VW71superbeetle committed:
Toasty

Added toast to most major action screens with positive success notifications. retrieving error messages is next.

demedukd committed:
Merge pull request #81 from 2103-USD/OrdersWork_20210911

Toasty changes

demedukd committed:
Updated Featured Products, Clear Products Filter

Featured Products:
Removed Arrows for slider , changed bottom buttons, added links to products
Added Students, Parents, Staff, Community links

Products
Added option to clear filter (included All option)
CSS
Updated Footer globally
Slight changes to navbar

demedukd committed:
CSS and more Product Changes

Featured Items
Removed Border-Radius

Products
Removed unborn inStock reference

VW71superbeetle committed:
Merge pull request #82 from 2103-USD/DatabaseObjects_20210912_2

Updated Featured Products, Clear Products Filter

VW71superbeetle committed:
Empty Cart button

Empty Cart button has been restored to the cart page

captnjak committed:
added toast styling

VW71superbeetle committed:
Register and Login notifications

Fix notifications for login/register failures, and implement toast notifications for all events.

VW71superbeetle committed:
Add total to DB order calls

Total field now returned as part of all getOrder type calls

VW71superbeetle committed:
console.log cleanup

VW71superbeetle committed:
Merge pull request #83 from 2103-USD/FrontEnd9-12

added toast styling

captnjak committed:
Merge pull request #84 from 2103-USD/OrdersWork_20210911

Username/login fixes, and new order totals for all order type calls

demedukd committed:
More CSS

demedukd committed:
Merge branch 'dev' into DatabaseObjects_20210912_2

VW71superbeetle committed:
Merge pull request #85 from 2103-USD/DatabaseObjects_20210912_2

More CSS

demedukd committed:
Extra App divs

and pinche footer

captnjak committed:
Merge pull request #86 from 2103-USD/DatabaseObjects_20210912_2

Extra App divs

VW71superbeetle committed:
Update orders.js

Fix aliasing

demedukd committed:
Merge pull request #87 from 2103-USD/OrdersWork_20210911

Update orders.js

VW71superbeetle committed:
Clear console logs

Clear console logs

VW71superbeetle committed:
add support for UUID

Required for stripe

VW71superbeetle committed:
Clear more console logs

Clearing out more console logs

VW71superbeetle committed:
Stripe!!!!

Initial stripe work, with success notification being returned.

captnjak committed:
Merge pull request #88 from 2103-USD/OrdersWork_20210911

Orders work 20210911

captnjak committed:
updated css and added MyAccount but needs work

captnjak committed:
added cancel button to admin edits

VW71superbeetle committed:
Update auth.js

VW71superbeetle committed:
Merge pull request #89 from 2103-USD/FrontEnd9-12.2

updated css and added MyAccount but needs work

captnjak committed:
CSS for Admin Product card

demedukd committed:
Merge pull request #90 from 2103-USD/FrontEnd9-12.3

CSS for Admin Product card

demedukd committed:
Added Search bar (Google redirect) and more CSS

CSS
Fixed footer. Now stays at bottom of page, but scrolling is required to view it.

NavBar
Added Google search

captnjak committed:
updated db/route/api USERS to accomodate MyAccount changes

VW71superbeetle committed:
CompleOrder functionality

CompleteOrder now stores the payment information in new fields, sets status to completed, and deducts the quantity from inStock.

Payment result messages now toast.

demedukd committed:
Merge pull request #92 from 2103-USD/OrdersWork_20210911

CompleOrder functionality

captnjak committed:
formatting on myaccount

VW71superbeetle committed:
Merge branch 'dev' into 20210912_3

VW71superbeetle committed:
Merge pull request #91 from 2103-USD/20210912_3

Added Search bar (Google redirect) and more CSS

VW71superbeetle committed:
Merge pull request #93 from 2103-USD/FrontEnd9-12.3

Front end9 12.3

VW71superbeetle committed:
Updated Index

Updated Index in proper alphabetical order for Derek

captnjak committed:
Merge pull request #94 from 2103-USD/OrdersWork_20210911

Updated Index

captnjak committed:
css changes and styling

captnjak committed:
removed opacity where necessary

captnjak committed:
added timestamp format using moment

VW71superbeetle committed:
Merge pull request #95 from 2103-USD/FrontEnd9-12.4

css changes and styling

VW71superbeetle committed:
Max qty in cart

Stop a user from purchasing more than is available at the moment.

VW71superbeetle committed:
Reorder Navbar

Keep all admin functions together to the right of generally available user functions

VW71superbeetle committed:
Update Order listings

Now includes payment information and formatted date

VW71superbeetle committed:
Update AdminOrderCard.jsx

Includes payment information

captnjak committed:
Merge pull request #96 from 2103-USD/OrdersWork_20210911

Orders work 20210911

demedukd committed:
Footer icon updates, Casing, etc

demedukd committed:
More CSS

Social link on products.

demedukd committed:
Delete SocialLinks.jsx

captnjak committed:
Merge pull request #97 from 2103-USD/DatabaseObjects_20210913

Footer icon updates, Casing, etc

demedukd committed:
Navbar fix

captnjak committed:
Merge pull request #98 from 2103-USD/DatabaseObjects_20210913

Navbar fix

VW71superbeetle committed:
Update GetMyOrders function

demedukd committed:
Merge pull request #99 from 2103-USD/OrdersWork_20210911

Update GetMyOrders function

VW71superbeetle committed:
oops

captnjak committed:
Merge pull request #100 from 2103-USD/OrdersWork_20210911

oops

VW71superbeetle committed:
Updated backend MyOrders

captnjak committed:
Merge pull request #101 from 2103-USD/OrdersWork_20210911

Updated backend MyOrders

captnjak committed:
Revert "Updated backend MyOrders"

demedukd committed:
Merge pull request #102 from 2103-USD/revert-101-OrdersWork_20210911

Revert "Updated backend MyOrders"

captnjak committed:
Revert "oops"

demedukd committed:
Merge pull request #103 from 2103-USD/revert-100-OrdersWork_20210911

Revert "oops"

captnjak committed:
Revert "Update GetMyOrders function"

demedukd committed:
Merge pull request #104 from 2103-USD/revert-99-OrdersWork_20210911

Revert "Update GetMyOrders function"

captnjak committed:
updated charSet on ProductCard

demedukd committed:
Merge pull request #105 from 2103-USD/FrontEnd9-13.2

updated charSet on ProductCard

captnjak committed:
updated order to be require user instead of admin

captnjak committed:
updated backend calls Joel wrote and frontend Orders

captnjak committed:
Updated Auth

updated the auth to point at Heroku

demedukd committed:
Merge pull request #106 from 2103-USD/FrontEnd9-13.2

Front end9 13.2

captnjak committed:
Revert "Front end9 13.2"

demedukd committed:
Merge pull request #107 from 2103-USD/revert-106-FrontEnd9-13.2

Revert "Front end9 13.2"
